dds/DCPS/WriterDataSampleList.inl

```diff
@@ -58,14 +59,14 @@ WriterDataSampleList::enqueue_tail(const DataSampleElement* sample)
   // changed to accommodate const-correctness throughout.
   DataSampleElement* mSample = const_cast<DataSampleElement*>(sample);
 
-  //sample->previous_writer_sample_ = 0;
-  //sample->next_writer_sample_ = 0;
+  mSample->next_writer_sample_ = 0;
```

Comment:
I'm wondering about why these lines were commented out. Is it possible anyone is using this function to enqueue multiple samples onto the tail by chaining them before passing them in (e.g. concatenating two lists)? If so, this would break that usage.

Reply:
This function is called in `WriteDataContainer` and I think the input is a single element. Maybe these lines were commented out by mistake?

Reply:
@simpsont-oci I don't think the multiple samples use case would be expected due to `++size_`
@sonndinh You could check the revision history for when they were commented out.

Reply:
The oldest commit I can find on GitHub was from Jul 03, 2014 (https://github.com/objectcomputing/OpenDDS/commit/304939ca0212b4ed75b2f215a86d085db3f7eee5#). Looks like these two lines were already commented out before the code was imported to GitHub.

Reply:
This repository has commits going back to the start of the project, before GitHub existed (probably?).  They were imported from SVN (and CVS before that).  It may be difficult to see where this came from due to renamed files.

Reply:
That may be a carry over from when DataSampleList was split up into WriterDataSampleList, SendStateDataSampleList, and InstanceDataSampleList, which I believe may have all been based off of the DataSampleList initially.  Might have been left in, correctly or incorrectly, as a note on the historical implementation in the DataSampleList.